Topics/Tech_Stacks/Tailwind.md

```diff
@@ -0,0 +1,101 @@
+2. The generated `tailwind.config.js` configuration file contains the default setup. Give Tailwind access to files by listing the template paths in the content section. For React/TypeScript web app, it can look like this:
+```js
+content: [
+    './pages/**/*.{ts,tsx}',
+    './components/**/*.{ts,tsx}',
+    './app/**/*.{ts,tsx}',
+    './src/**/*.{ts,tsx}',
+  ],
+```
+
+3. Add the following to the global CSS file:
+```css
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+```
+
+Now we're set!
+
+## Usage
+
+### The Basics
+Each Tailwind class is focused on a specific responsibility. To use a class, add it to the `class` or `className` (when using React) attribute of the element/component. For example, use the following Tailwind classes to create a div that
+- occupies the full width of the screen
+- allow scrolling within
+- has a sky blue background
+- has large text
+
+```jsx
+<div className="w-screen overflow-scroll bg-sky-400 text-lg">
+    {/* Some content */} 
+</div>
+```
+
+Start [here](https://tailwindcss.com/docs/aspect-ratio) to learn about all Tailwind classes.
+
+### Responsive Design
+Tailwind is powerful for creating a suitable experience across different devices.
+
+#### Targeted Screen Sizes
+To set a style for specific screen sizes, Tailwind offers breakpoints. These apply a style only if the screen width exceeds the breakpoint's minimum width. Breakpoints are an alternative to the traditional CSS `@media` queries. 
+
+Breakpoints:
+- 'sm' - 640px
+- 'md' - 768px
+- 'lg' - 1024px
+- 'xl' - 1280px
+- '2xl' - 1536px
+
+To denote a breakpoint, add the prefix before a class name to conditionally apply: `<breakpoint-prefix>:<class-name>` 
+
+For example, to make the height of a div 100%, on screens more than 1024 pixels wide:
+```jsx
+<div className="lg:h-full">
+    {/* Some content */} 
+</div>
+```
+
+> **NOTE:** A common mistake is to use 'sm' to target mobile devices. Mobile sizes have the smallest widths, so their styles should **not** have a breakpoint prefix. 
+
+
+## Flexibility
+When using Tailwind CSS classes, developers are not restricted to a component structure and inherited styles that don't match their project. This can be problematic when using templates from component libraries such as [Bootstrap](https://getbootstrap.com). This flexibility makes Tailwind more customizable for different types of projects.
+
+Since each Tailwind class has atomic reponsibilities, developers avoid issues where a component inherits unknown CSS styles. 
+
+Overall, Tailwind minimizes redundant CSS code and shortens CSS files.
+
+## Limitations
+If accustomed to writing CSS, it can take some time to get used to using the equivalent Tailwind classes. 
+
+Also, some projects may not need Tailwind. Using the framework requires additional overhead, which may not be worth it for small projects.
+
+It may take more time to complete projects, since Tailwind prioritizes styling control over pre-made components.
+
+
+## Resources
```

Comment:
I had an idea for another resource you could consider adding. You could consider briefly introducing the idea of states (ie. hover states, focus states, etc) and/or linking this page here so readers can learn more: https://tailwindcss.com/docs/hover-focus-and-other-states . I found these states to be a nice feature of Tailwind and helped me style components in my project.

Reply:
Thanks, I forgot to include these. State declarations are definitely useful. I will include it in References, since they're quite similar to their CSS counterparts and don't directly apply to the Responsiveness subject.

Reply:
Makes sense!